cnf-certification-test/accesscontrol/suite.go

```diff
@@ -410,44 +410,58 @@ func testPodRoleBindings(env *provider.TestEnvironment) {
 			continue
 		}
 
-		// Get any rolebindings that do not belong to the pod namespace.
-		roleBindings, err := rbac.GetRoleBindings(put.Namespace, put.Spec.ServiceAccountName)
-		if err != nil {
-			failedPods = append(failedPods, put.Name)
-		}
+		logrus.Infof("%s has a serviceAccountName: %s, checking role bindings.", put.String(), put.Spec.ServiceAccountName)
+
+		// Loop through the rolebindings and check if they are from another namespace
+		for rbIndex := range env.RoleBindings {
+			found := false
+			for _, subject := range env.RoleBindings[rbIndex].Subjects {
+				// Match the subject service account with the name of the pod's service account.
+				// Determine if that service account is not in the same namespace as the pod.
+				if subject.Kind != "ServiceAccount" || subject.Name != put.Spec.ServiceAccountName {
+					continue
+				}
 
-		if len(roleBindings) > 0 {
-			logrus.Warnf("Pod: %s/%s has the following role bindings: %s", put.Namespace, put.Name, roleBindings)
-			tnf.ClaimFilePrintf("Pod: %s/%s has the following role bindings: %s", put.Namespace, put.Name, roleBindings)
-			failedPods = append(failedPods, put.Name)
+				if rbac.IsRoleBindingOutOfNamespace(put.Namespace, put.Spec.ServiceAccountName, env.RoleBindings[rbIndex].Namespace, env.RoleBindings[rbIndex].Name) {
```

Comment:
Are these helper functions really needed? My suggestion:
```
		RB_LOOP:
		for rbIndex := range env.RoleBindings {
			if env.RoleBindings[rbIndex].Namespace == put.Namespace {
				continue
			}

			for _, subject := range env.RoleBindings[rbIndex].Subjects {
				if subject.Kind == rbacv1.ServiceAccountKind && subject.Namespace == put.Namespace && subject.Name == put.Spec.ServiceAccountName {
					logrus.Warnf("Pod: %s/%s has the following role bindings that do not live in the same namespace: %s", put.Namespace, put.Name, env.RoleBindings[rbIndex].Name)
					tnf.ClaimFilePrintf("Pod: %s/%s has the following role bindings that do not live in the same namespace: %s", put.Namespace, put.Name, env.RoleBindings)
					failedPods = append(failedPods, put.String())
					break RB_LOOP
				}
			}
		}
```
Also note that in that helper func `roleBindingOutOfNamespace()` you're using the RoleBinding name to compare it against the ServiceAccount name, but I believe that's wrong: you don't that RoleBinding name here. The ServiceAccount name should be only compared with the subject.Name.

Reply:
Okay this makes sense.  Agreed the old carryover code from `roleBindingOutOfNamespace` is not correct I will remove it and use the suggestion above.